### ISSUE TITLE
Exception caught in Api Request NoSuchMethodError

### ISSUE THREAD

**Issue by rj-since-2000** [CCExtractor/rutorrent-flutter#90]
```
Class 'List<dynamic>' has no instance getter 'keys'.
I/flutter (31423): Receiver: Instance(length:0) of '_GrowableList'
I/flutter (31423): Tried calling: keys
```
There is no check for torrent list response from ApiRequest before accessing its path keys so when the torrent list is empty still it tries to access 'keys' getter which doesn't exist.

@harchani-ritik I propose to add a check for elements/size of response torrent list before accessing any of it's method to prevent this exception.

**Comment by nb9960** [comment]
> ```
> Class 'List<dynamic>' has no instance getter 'keys'.
> I/flutter (31423): Receiver: Instance(length:0) of '_GrowableList'
> I/flutter (31423): Tried calling: keys
> ```
> 
> There is no check for torrent list response from ApiRequest before accessing its path keys so when the torrent list is empty still it tries to access 'keys' getter which doesn't exist.
> 
> @harchani-ritik I propose to add a check for elements/size of response torrent list before accessing any of it's method to prevent this exception.

Same issue occurs in history screen too, in case of empty list.

**Comment by pratikbaid3** [comment]
@rj-since-2000 I guess you can create one single issue detailing all the error handling issues and I guess @tusharojha is already working on the error handling PR

**Comment by pratikbaid3** [comment]
#90 #91 #89 #73 #83 are all related to error handling.

**Comment by tusharojha** [comment]
Not all of these but yes majorly the issues related to error handling will be resolved in #64. 
Actually, I am waiting for @harchani-ritik  for a review after that I will make the suggested changes.

**Comment by pratikbaid3** [comment]
@tusharojha Could you mention the issues that your PR might resolve so that @rj-since-2000 can work on the others.

**Comment by rj-since-2000** [comment]
@pratikbaid3 @tusharojha 
PR #64 is moreover related to errors related to server side response and adding all these error handling to it may end up having too much for a single PR. I think we can have a single issue for all these [issues](https://github.com/CCExtractor/rutorrent-flutter/issues/90#issuecomment-798888442)

**Comment by tusharojha** [comment]
> Could you mention the issues that your PR might resolve so that @rj-since-2000 can work on the both

Sure, #90 #91 only.

**Comment by rj-since-2000** [comment]
@tusharojha If you have solved this issue in #64 should I close this ?
